Source/Core/Core/LEDWin.cpp

```diff
@@ -0,0 +1,284 @@
+	{ "LWIN", LogiLed::KeyName::LEFT_WINDOWS },
+	{ "RWIN", LogiLed::KeyName::RIGHT_WINDOWS },
+	{ "APPS", LogiLed::KeyName::APPLICATION_SELECT },
+	// F-keys
+	{ "F1", LogiLed::KeyName::F1 },
+	{ "F2", LogiLed::KeyName::F2 },
+	{ "F3", LogiLed::KeyName::F3 },
+	{ "F4", LogiLed::KeyName::F4 },
+	{ "F5", LogiLed::KeyName::F5 },
+	{ "F6", LogiLed::KeyName::F6 },
+	{ "F7", LogiLed::KeyName::F7 },
+	{ "F8", LogiLed::KeyName::F8 },
+	{ "F9", LogiLed::KeyName::F9 },
+	{ "F10", LogiLed::KeyName::F10 },
+	{ "F11", LogiLed::KeyName::F11 },
+	{ "F12", LogiLed::KeyName::F12 },
+	// other keys
+	{ "ESCAPE", LogiLed::KeyName::ESC },
+	{ "INSERT", LogiLed::KeyName::INSERT },
+	{ "DELETE", LogiLed::KeyName::KEYBOARD_DELETE },
+	{ "HOME", LogiLed::KeyName::HOME },
+	{ "END", LogiLed::KeyName::END },
+	{ "PRIOR", LogiLed::KeyName::PAGE_UP },
+	{ "NEXT", LogiLed::KeyName::PAGE_DOWN },
+	{ "SYSRQ", LogiLed::KeyName::PRINT_SCREEN },
+	{ "SCROLL", LogiLed::KeyName::SCROLL_LOCK },
+	{ "PAUSE", LogiLed::KeyName::PAUSE_BREAK },
+};
+
+// are we ready to call LogiLed* functions?
+static bool s_initialized = false;
+
+void Initialize()
+{
+	// see if we can init the Logitech Gaming LED SDK. If not, just silently drop out.
+	// this needs to be done before calling any LogiLed* functions; and some of them
+	// even require some delay between calls in order to register properly.
+	s_initialized = LogiLedInit();
+
+	if (s_initialized)
+	{
+		int major;
+		int minor;
+		int build;
+		if (LogiLedGetSdkVersion(&major, &minor, &build))
+			NOTICE_LOG(PAD, "Using Logitech Gaming LED SDK v%d.%d.%d", major, minor, build);
+	}
+}
+
+void Start()
+{
+	if (!s_initialized)
+		return;
+
+	// remember what the current lighting looks like.
+	// this actually returns success, but we'll just overwrite the layout regardless
+	// (since it will be reset at latest when shutdown is called).
+	if (!LogiLedSaveCurrentLighting())
+		WARN_LOG(PAD, "Failed to save current Lighting state");
+
+	// it seems that LogiLedSaveCurrentLighting has a slight delay and needs to finish before
+	// doing anything else with it. Otherwise it might store the new state instead of the old
+	// one if we call LogiLedSetLighting(ForKeyName) too quickly.
+	// 20ms seemed to work fine during testing, 10ms is too low. Values up to 50 or even 100ms
+	// are not really noticable and could be used in case 20 doesn't work everywhere.
+	Common::SleepCurrentThread(20);
+
+	// only talk to per-key devices for now; others won't make much sense
+	// due to their limited possibilities of displaying control state.
+	LogiLedSetTargetDevice(LOGI_DEVICETYPE_PERKEY_RGB);
+	// black out all keys initially
+	LogiLedSetLighting(0, 0, 0);
+
+	InputConfig* const pad_plugin = Pad::GetConfig();
+	// grab the Player 1 controller (TODO: what about other local players?)
+	ControllerEmu* const controller = pad_plugin->controllers[0];
+
+#define COLOR_PERCENT(c) ((c).red * 100 / 0xFF), ((c).green * 100 / 0xFF), ((c).blue * 100 / 0xFF)
```

Comment:
Mostly for being too lazy to type the ```s_colors[idx].col``` part all the time. Had multiple uses initially, can drop it.